sympy/functions/combinatorial/factorials.py

```diff
@@ -742,8 +742,11 @@ def eval(cls, x, k):
                                             range(1, abs(int(k)) + 1), 1)
 
     def _eval_rewrite_as_gamma(self, x, k, **kwargs):
-        from sympy import gamma
-        return (-1)**k*gamma(k - x) / gamma(-x)
+        from sympy import gamma, Piecewise
```

Comment:
Based on our disucussions and work in #18960, I don't think Piecewise should be returned. If the assumptions are insufficient to allow unambiguous rewriting as `gamma` then nothing should be returned:
```python
if x.is_negative:
    return (-1)**k*gamma(k - x) / gamma(-x)
if x.is_nonnegative:
    return gamma(x + 1) / gamma(x - k + 1)
```

Reply:
This is failing the **rst doctests in concrete**:

```
**********************************************************************
File "sympy\doc\src\modules\concrete.rst", line 31, in concrete.rst
Failed example:
    ff(n, k).is_hypergeometric(n)
Expected:
    True
Got:
    False
**********************************************************************
File "sympy\doc\src\modules\concrete.rst", line 46, in concrete.rst
Failed example:
    ff(n, k).is_hypergeometric(k)
Expected:
    True
Got:
    False
**********************************************************************
File "sympy\doc\src\modules\concrete.rst", line 58, in concrete.rst
Failed example:
    ff(n-1, k+1).is_hypergeometric(n)
Expected:
    True
Got:
    False
**********************************************************************
1 items had failures:
   3 of  22 in concrete.rst
***Test Failed*** 3 failures.
doc\src\modules\concrete.rst [22]                                            [3]
```

Reply:
The problem is:

**Earlier:**
```
In [1]: ff(n, k)
Out[1]: (-1)**k*gamma(k - n) / gamma(-n)
```

**Now:**
```
In [1]: ff(n, k)
Out[1]: FallingFactorial(n, k)
```